### PR TITLE
expression: implement vectorized evaluation for `builtinArithmeticPlusIntSig` 

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Implement vectorized evaluation for `builtinArithmeticPlusIntSig` 

### What is changed and how it works?
Almost 7x faster than before:
```
BenchmarkVectorizedBuiltinFunc/builtinArithmeticPlusIntSig-VecBuiltinFunc-12              500000              2832 ns/op
BenchmarkVectorizedBuiltinFunc/builtinArithmeticPlusIntSig-NonVecBuiltinFunc-12           100000             19792 ns/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_arithmetic_vec.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	if err := s.args[1].VecEvalInt(s.ctx, input, buf); err != nil {
+		return err
+	}
+
+	isLHSUnsigned := mysql.HasUnsignedFlag(s.args[0].GetType().Flag)
+	isRHSUnsigned := mysql.HasUnsignedFlag(s.args[1].GetType().Flag)
+	as := result.Int64s()
+	bs := buf.Int64s()
+	switch {
+	case isLHSUnsigned && isRHSUnsigned:
```
Comment (b41sh) [inline expression/builtin_arithmetic_vec.go]: Looks like some duplication of code

**Comment by zz-jason** [comment]
to https://github.com/pingcap/tidb/issues/12058

File: expression/builtin_arithmetic_vec_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	"testing"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/tidb/types"
+)
+
+var vecBuiltinArithmeticCases = map[string][]vecExprBenchCase{
+	ast.Plus: {
+		{types.ETInt, []types.EvalType{types.ETInt, types.ETInt}, []dataGenerator{&rangeInt64Gener{0, 100000}, &rangeInt64Gener{0, 100000}}},
```
Comment (SunRunAway) [inline expression/builtin_arithmetic_vec_test.go]: Please add struct keys here for further compatibility.

**Pull request closed without merging**